internal/api/courses/users/drop.go

```diff
@@ -0,0 +1,41 @@
+package users
+
+import (
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/model"
+)
+
+type DropRequest struct {
+	core.APIRequestCourseUserContext
+	core.MinCourseRoleAdmin
+
+	TargetCourseUser core.TargetCourseUser `json:"target-email"`
+}
+
+type DropResponse struct {
+	FoundUser bool `json:"found-user"`
+}
+
+func HandleDrop(request *DropRequest) (*DropResponse, *core.APIError) {
+	response := DropResponse{}
+
+	if !request.TargetCourseUser.Found {
+		return &response, nil
+	}
+
+	response.FoundUser = true
+
+	if (request.ServerUser.Role < model.ServerRoleAdmin) && (request.TargetCourseUser.User.Role >= request.User.Role) {
```

Comment:
Make a function (checkPermissions()?) that handles this permissions check.
Then once there, handle this in two checks:
 1. Server admins can do what they want (return early). Put a comment here that we are letting a server admin drop a server owner, but that we don't really care since it does not change things on the server-level.
 2. If you are not a server admin, you need high enough relative permissions.

The code you have here is correct (and I would probably let it in if it was the only change I was considering for this PR),
but the nice thing about having it in another function is just how clear you can make it to other developers what is going on in it.
And since this is a potential security task, we want to be as clear as possible.